openedx/features/subscriptions/models.py

```diff
@@ -0,0 +1,89 @@
+import logging
+
+from datetime import date
+
+from django.contrib.auth.models import User
+from django.db import models
+from django.db.models import Q
+from model_utils.models import TimeStampedModel
+
+from student.models import CourseEnrollment
+
+class UserSubscription(TimeStampedModel):
+    """
+    The "UserSubscription" model.
+    """
+    user = models.ForeignKey(User, on_delete=models.CASCADE)
+    subscription_id = models.PositiveIntegerField(db_index=True)
+    expiration_date = models.DateField(default=None, null=True, blank=True)
+
+    LIMITED_ACCESS = 'limited-access'
+    FULL_ACCESS_COURSES = 'full-access-courses'
+    FULL_ACCESS_TIME_PERIOD = 'full-access-time-period'
+    LIFETIME_ACCESS = 'lifetime-access'
+
+    SUBSCRIPTION_TYPE_CHOICES = (
+        (LIMITED_ACCESS, 'Limited Access', ),
+        (FULL_ACCESS_COURSES, 'Full Access (Courses)', ),
+        (FULL_ACCESS_TIME_PERIOD, 'Full Access (Time period)', ),
+        (LIFETIME_ACCESS, 'Lifetime Access', ),
+    )
+
+    subscription_type = models.CharField(choices=SUBSCRIPTION_TYPE_CHOICES, max_length=30)
+    number_of_courses = models.PositiveIntegerField(null=True, blank=True)
```

Comment:
I think we can rename this field to `max_courses`
```python
max_allowed_courses = models.PositiveIntegerField(
    help_text=_(
        'Maximum number of courses allowed to enroll in paid tracks using the subscription.'
    )
    null=True, 
    blank=True
)
```

Reply:
done